ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/SearchIndexTool.java

```diff
@@ -120,8 +123,16 @@ public <T> void run(Map<String, String> parameters, ActionListener<T> listener)
         try {
             String input = parameters.get(INPUT_FIELD);
```

Comment:
if (input != null && !input.isEmpty()) {
                try {
                    jsonObject = GSON.fromJson(input, JsonObject.class);
                } catch (JsonSyntaxException e) {
                    log.warn("Invalid JSON input: {}", input, e);
                }
            }
consider adding this null check and error handling?